libs/challenge-registry/challenge-search/src/lib/challenge-search.component.ts

```diff
@@ -134,12 +151,28 @@ export class ChallengeSearchComponent implements OnInit, AfterContentInit {
   }
 
   ngAfterContentInit(): void {
+    this.searchTerms
+      .pipe(
+        debounceTime(1000), // wait for 1s
```

Comment:
I did some research in the past and 400 or 450 ms were recommended for the debounce time to get a responsive behavior.

```suggestion
        debounceTime(400),
```